src/main/kotlin/org/rust/lang/core/lexer/RustStringLiteralLexer.kt

```diff
@@ -0,0 +1,137 @@
+package org.rust.lang.core.lexer
+
+import com.intellij.openapi.util.text.StringUtil
+import com.intellij.psi.StringEscapesTokenTypes.*
+import com.intellij.psi.tree.IElementType
+
+private const val BYTE_ESCAPE_LENGTH = "\\x00".length
+private const val UNICODE_ESCAPE_MIN_LENGTH = "\\u{0}".length
+private const val UNICODE_ESCAPE_MAX_LENGTH = "\\u{000000}".length
+
+// TODO: Maybe implement lexing of prefixes and suffixes?
```

Comment:
I'm not sure whether it makes sense to implement this. Distinguished prefixes/suffixes may be useful for making rainbow colour schemes, but currently I don't see any usage for them, unless in some wild inspections.